plugins/org.python.pydev.ast/src/org/python/pydev/ast/interpreter_managers/InterpreterInfo.java

```diff
@@ -353,34 +354,46 @@ public static InterpreterInfo fromString(String received, boolean askUserInOutPa
 
                         } else if ("executable".equals(name)) {
                             infoExecutable = data;
+                            if (userSpecifiedExecutable != null) {
+                                // Determine if the interpreter's self-reported executable name is the same as what was
+                                // requested, if there is an extra prefix then that needs to also be prefixed to every
+                                // lib that is returned
+                                if (!userSpecifiedExecutable.equals(infoExecutable)
```

Comment:
Can you add here more details on when and why this happens?

It seems like the user selected

`/usr/bin/python`

and then got an interpreter as

`/foo/usr/bin/python`

and then we add the /foo to each entry... is that it?

Note that this seems a bit strange to me... why didn't interpreterInfo.py already yield those that way? What happens if the user just specified `python` and then received `/usr/bin/python`, would we prefix `/usr/bin` to all PYTHONPATH entries? Isn't there a better way to detect that structure from the python side in `interpreterInfo.py` (where we have a running python instance)?

Reply:
Actually, I didn't previously see your comment saying that python is sandboxed and doesn't know about it... 

Do you think there's another way to detect it?

I think that just seeing that it was a prefix may be a bit brittle... although if that's the only way, we should definitely first see if the directories exist before adding the prefix so that if the user specifies only `python` and then gets `/usr/bin/python` we don't prefix the libraries with `/usr/bin`.

Also, just as a note, the interpreter doesn't have to be the real interpreter -- in a number of cases it can be just a .sh/.bat which configures something and then turns on to execute the real Python, so, just relying on the path may be a bit brittle (as the user could hit that on other cases).